mesonbuild/modules/simd.py

```diff
@@ -74,39 +74,46 @@ def __init__(self, interpreter: Interpreter):
     def check(self, state: ModuleState, args: T.Tuple[str], kwargs: CheckKw) -> T.List[T.Union[T.List[build.StaticLibrary], build.ConfigurationData]]:
-        result: T.List[build.StaticLibrary] = []
         if 'sources' in kwargs:
             raise mesonlib.MesonException('SIMD module does not support the "sources" keyword')
+
+        result: T.List[build.StaticLibrary] = []
         prefix = args[0]
-        basic_kwargs = {}
-        for key, value in kwargs.items():
-            if key not in ISETS and key != 'compiler':
-                basic_kwargs[key] = value
         compiler = kwargs['compiler']
         conf = build.ConfigurationData()
+
+        local_keys = set((*ISETS, 'compiler'))
+        static_lib_kwargs = T.cast('kwtypes.StaticLibrary', {k: v for k, v in kwargs.items() if k not in local_keys})
+
         for iset in ISETS:
-            if iset not in kwargs:
+            sources = kwargs[iset] # type: ignore
+
+            cargs = compiler.get_instruction_set_args(iset)
+            if cargs is None:
+                mlog.log(f'Compiler supports {iset}:', mlog.red('NO'))
                 continue
-            iset_fname = kwargs[iset] # Might also be an array or Files. static_library will validate.
-            args = compiler.get_instruction_set_args(iset)
-            if args is None:
-                mlog.log('Compiler supports %s:' % iset, mlog.red('NO'))
+
+            if not compiler.has_multi_arguments(cargs, state.environment)[0]:
+                mlog.log(f'Compiler supports {iset}:', mlog.red('NO'))
                 continue
-            if args:
-                if not compiler.has_multi_arguments(args, state.environment)[0]:
-                    mlog.log('Compiler supports %s:' % iset, mlog.red('NO'))
-                    continue
-            mlog.log('Compiler supports %s:' % iset, mlog.green('YES'))
-            conf.values['HAVE_' + iset.upper()] = ('1', 'Compiler supports %s.' % iset)
-            libname = prefix + '_' + iset
-            lib_kwargs = {'sources': iset_fname,
-                          }
-            lib_kwargs.update(basic_kwargs)
-            langarg_key = compiler.get_language() + '_args'
-            old_lang_args = mesonlib.extract_as_list(lib_kwargs, langarg_key)
-            all_lang_args = old_lang_args + args
-            lib_kwargs[langarg_key] = all_lang_args
-            result.append(self.interpreter.func_static_lib(None, [libname], lib_kwargs))
+
+            mlog.log(f'Compiler supports {iset}:', mlog.green('YES'))
+            conf.values['HAVE_' + iset.upper()] = ('1', f'Compiler supports {iset}.')
+
+            my_name = f'{prefix}_{iset}'
+            my_kwargs = static_lib_kwargs.copy()
+            my_kwargs['sources'] = sources # type: ignore
+
+            # Add compile args we derived above to those the user provided us
+            lang_args_key = compiler.get_language() + '_args'
```

Comment:
As previously mentioned at https://github.com/mesonbuild/meson/pull/12104#discussion_r1290489180

> In general, this commit seems to have leaned very heavily into avoiding cases where git can recognize previous content in the diff.


Some of the decisions in this "clean up the code" confuse me quite a bit. As an example, why rename `langarg_key` to `lang_args_key`? Are the new variable names better? Keeping the old ones should allow git to show the actual semantic changes.

Reply:
Well it is documented as `<lang>_args`, not `<lang>_arg` (`c_args`, not `c_arg`

Reply:
Plus it is `old_lang_args`, not `old_langarg`. If anything, this particular rename is more consistent

Reply:
Anyway, removed the `my` prefix in favor of `lib`

Reply:
```diff
-libname = prefix + '_' + iset
+my_name = f'{prefix}_{iset}'
```
This name change feels very... Perl Programming Language.

But also, I don't actually like f-strings in such a case -- it actively feels harder to read when the f-string is almost entirely brackets denoting variables.

Reply:
No idea how that is more readable, but ok, here you go in the next push

Reply:
> Anyway, removed the `my` prefix in favor of `lib`

This is not what I asked for. :(

```diff
-libname = prefix + '_' + iset
+lib_name = prefix + '_' + iset
```

> If anything, this particular rename is more consistent

It's such an incredibly subjective "rename is more consistent", and that's the root of my objection. I'm not a fan of "submit a PR that contains blocks of 20 or 30 lines that are just one big hunk of removed followed by one big hunk of added, because a bunch of variables got renamed".

And adding an underscore to the variable name fails at my target objective, which was to make that line show up as unchanged and divide up the `git log -p` view.

Reply:
> Well it is documented as `<lang>_args`, not `<lang>_arg` (`c_args`, not `c_arg`

I am not sure why the internal python code names need to match the precise spelling, underscore style, and pluralization of the meson.build form. It is simply not important! But being able to `git diff` is important, and that means that bikesheddy variable renames are a bad thing when they mean that the entire file is a solid wall of red and green.

Reply:
```diff
diff --git a/mesonbuild/modules/simd.py b/mesonbuild/modules/simd.py
index 2be4e9164..0a5762b46 100644
--- a/mesonbuild/modules/simd.py
+++ b/mesonbuild/modules/simd.py
@@ -88,29 +88,29 @@ class SimdModule(ExtensionModule):
         for iset in ISETS:
             sources = kwargs[iset] # type: ignore
 
-            cargs = compiler.get_instruction_set_args(iset)
-            if cargs is None:
+            args = compiler.get_instruction_set_args(iset)
+            if args is None:
                 mlog.log(f'Compiler supports {iset}:', mlog.red('NO'))
                 continue
 
-            if not compiler.has_multi_arguments(cargs, state.environment)[0]:
+            if not compiler.has_multi_arguments(args, state.environment)[0]:
                 mlog.log(f'Compiler supports {iset}:', mlog.red('NO'))
                 continue
 
             mlog.log(f'Compiler supports {iset}:', mlog.green('YES'))
             conf.values['HAVE_' + iset.upper()] = ('1', f'Compiler supports {iset}.')
 
-            lib_name = prefix + '_' + iset
+            libname = prefix + '_' + iset
             lib_kwargs = static_lib_kwargs.copy()
             lib_kwargs['sources'] = sources # type: ignore
 
             # Add compile args we derived above to those the user provided us
-            lang_args_key = compiler.get_language() + '_args'
-            old_lang_args = mesonlib.extract_as_list(lib_kwargs, lang_args_key) # type: ignore
-            all_lang_args = old_lang_args + cargs
-            lib_kwargs[lang_args_key] = all_lang_args # type: ignore
+            langarg_key = compiler.get_language() + '_args'
+            old_lang_args = mesonlib.extract_as_list(lib_kwargs, langarg_key) # type: ignore
+            all_lang_args = old_lang_args + args
+            lib_kwargs[langarg_key] = all_lang_args # type: ignore
 
-            lib = self.interpreter.build_target(state.current_node, (lib_name, []), lib_kwargs, build.StaticLibrary)
+            lib = self.interpreter.build_target(state.current_node, (libname, []), lib_kwargs, build.StaticLibrary)
 
             result.append(lib)
 
```

This makes it significantly easier to follow the changes that were made, the commit still achieves everything it seems to be trying to achieve, but without unnecessary variable renames it also becomes *incredibly* easier to actually... review.

As I pointed out in August.

Reply:
```diff
-libname = prefix + '_' + iset
+my_name = f'{prefix}_{iset}'
```

I'm sorry, what. This is *exactly* the case that f-strings make more readable.

Reply:
It's easier to read code when the syntax highlighting for variables is replaced by syntax highlighting for a string? Or it's easier to read code when you're reading strings that are mostly code, but with a single character of an actual string in the middle?

I think f-strings make things more readable when the majority of the string is, well, a string, and the context switch of scanning over to the right in order to find out what `{}` or `%s` is, is eliminated. Reducing context switches is fantastic! Requiring me to give up syntax highlighting isn't fantastic, but in simple cases it's a highly worthwhile tradeoff.

Reply:
This is what I see. I am curious now what you see in your theme.

![image](https://github.com/mesonbuild/meson/assets/11936772/1a018236-823a-41d9-a971-c0bbfbd56b35)

Reply:
Mine highlights inside the f-string as well.